src/main_gate.rs

```diff
@@ -233,6 +233,22 @@ pub struct MainGateConfig<const T: usize> {
     pub(crate) rc: Column<Fixed>,
 }
 
+impl<const T: usize> MainGateConfig<T> {
+    pub fn name_columns<F: PrimeField>(&self, region: &mut Region<'_, F>) {
```

Comment:
what is the purpose of this function?

Reply:
If the main gate expression does not fit, this function will name each column in the error message.

Reply:
Don't quite get "does not fit" here. Do you mean this function will print the column names in error message?
I remember the annotation `| | "some_name"` in halo2's code such as in `fn assign_advice` didn't print out any user defined string.